mcap_vendor/CMakeLists.txt

```diff
@@ -35,48 +34,37 @@ macro(build_mcap_vendor)
   file(GLOB _lz4_srcs
     ${lz4_SOURCE_DIR}/lib/*.c)
 
-  add_library(
-    mcap SHARED
+  add_library(${PROJECT_NAME} SHARED
     src/main.cpp
     ${_lz4_srcs}
   )
 
   set(_mcap_include_dir ${mcap_SOURCE_DIR}/cpp/mcap/include)
 
-  target_include_directories(mcap SYSTEM PRIVATE
+  target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE
     ${lz4_SOURCE_DIR}/lib
   )
-  target_include_directories(mcap SYSTEM PUBLIC
+  target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
     "$<BUILD_INTERFACE:${_mcap_include_dir}>"
     "$<INSTALL_INTERFACE:include>"
   )
-  ament_target_dependencies(mcap zstd)
+  ament_target_dependencies(${PROJECT_NAME} zstd)
 
   install(
-    DIRECTORY
-      ${_mcap_include_dir}/mcap
-    DESTINATION
-      ${CMAKE_INSTALL_PREFIX}/include
+    DIRECTORY ${_mcap_include_dir}/mcap
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include
```

Comment:
While we are in here, we may as well make this follow our new recommendations to put this in a sub-sub-directory.  You can change this to:

```
DESTINATION include/${PROJECT_NAME}
```

(I believe the CMAKE_INSTALL_PREFIX is implicit, and we should add ${PROJECT_NAME} to the end)

Reply:
@clalancette I am not sure that I am correctly understand how do you want it to be with suggestion to replace it with `DESTINATION include/${PROJECT_NAME}`.
Do you want it to be the path as `install/mcap_vendor/include/mcap_vendor/` and then header files from `mcap` ?
If so, this is actually not what we want for this package.

Currently we have destination path as `install/mcap_vendor/include/mcap/` and then header files from `mcap`. The rational for that is that include files not from `mcap_vendor` package itself but rather from external `mcap` C++ project. And we want it to be clear and consistent.

Reply:
Or may be you want the path to be as `install/mcap_vendor/include/mcap_vendor/mcap/` and then header files from `mcap`. This is possible but looks odd and I don't see a rational for this.

Reply:
> Or may be you want the path to be as install/mcap_vendor/include/mcap_vendor/mcap/ and then header files from mcap. This is possible but looks odd and I don't see a rational for this.

This is what I am going for.

There is a good rationale for it; see https://github.com/ros2/ros2/issues/1150 for the gory details.  In short, if you look in `/opt/ros/rolling/include`, you'll see that every single one of the includes in the core packages lives in a sub-sub-directory so that we don't accidentally pull in the wrong header file during underlay/overlay processing.  I think we should do the same here.

Reply:
@clalancette Ok I changed it t o be under the `${PROJECT_NAME}` i.e. `install/mcap_vendor/include/mcap_vendor/mcap/`

Reply:
@clalancette Apparently it doesn't work this way. Build fails with error message that `mcap/mcap.hpp` not found.
I guess will need to change include to point to the `mcap_vendor/mcap/mcap.hpp`.  
I will try  to debug it locally.

Reply:
Fixed it as amend commit